roles/deploy-aws-infra/tasks/main.yml

```diff
@@ -0,0 +1,343 @@
+      - proto: tcp
+        from_port: 53
+        to_port: 53
+        cidr_ip: 0.0.0.0/0
+      - proto: udp
+        from_port: 4789
+        to_port: 4789
+        cidr_ip: 0.0.0.0/0
+  when: aws_create_vpc
+  register: ocp_master_sg
+
+- name: Create OCP Node Security Group for the new VPC
+  ec2_group:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    name: "labs-ocp-node"
+    description: "Labs OCP Node"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    state: "present"
+    tags:
+      env_id: "{{ env_id }}"
+    rules:
+      - proto: udp
+        from_port: 4789
+        to_port: 4789
+        cidr_ip: 0.0.0.0/0
+      - proto: tcp
+        from_port: 10250
+        to_port: 10250
+        cidr_ip: 0.0.0.0/0
+  when: aws_create_vpc
+  register: ocp_node_sg
+
+- name: Create OCP Infra Security Group for the new VPC
+  ec2_group:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    name: "labs-ocp-infra"
+    description: "Labs OCP Infra"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    state: "present"
+    tags:
+      env_id: "{{ env_id }}"
+    rules:
+      - proto: tcp
+        from_port: 80
+        to_port: 80
+        cidr_ip: 0.0.0.0/0
+      - proto: tcp
+        from_port: 443
+        to_port: 443
+        cidr_ip: 0.0.0.0/0
+  when: aws_create_vpc
+  register: ocp_infra_sg
+
+- name: Create an Internet Gateway for the new VPC
+  ec2_vpc_igw:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    state: present
+    tags:
+      env_id: "{{ env_id }}"
+  when: aws_create_vpc
+  register: new_gw
+
+- name: Create Gateway subnet route table for the new VPC
+  ec2_vpc_route_table:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    tags:
+      Name: "{{ env_id }}-route-table-gw"
+      env_id: "{{ env_id }}"
+    subnets:
+      - "{{ new_subnets.subnet.id }}"
+    routes:
+      - dest: "0.0.0.0/0"
+        gateway_id: "{{ new_gw.gateway_id }}"
+
+# Create local facts with new subnet id
+- name: Register new created objects as facts
+  set_fact:
+    aws_subnet: "{{ new_subnets.subnet.id }}"
+
+- name: Create Master and etcd instances
+  ec2:
+    image: "{{ aws_image_name }}"
+    instance_type: m4.xlarge
+    group: "{{ aws_master_sgroups }}"
+    termination_protection: yes
+    key_name: "{{ aws_key_name }}"
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    region: "{{ aws_region }}"
+    vpc_subnet_id: "{{ aws_subnet }}"
+    exact_count: 1
+    count_tag:
+      Name: "{{ env_id }}-ocp-master-{{ item }}"
+    instance_tags:
+      Name: "{{ env_id }}-ocp-master-{{ item }}"
+      user: sre
+      role: ocp
+      group: masters_aws
```

Comment:
Let's include the `env_id` here to avoid name clashes

Reply:
You mean including a tag called 'env_id'?

Reply:
No, the variable/fact called `env_id` - i.e.: something like `"{{ 'masters_' + env_id }}"`

Reply:
@oybed I think I don't get this one. Where do you see a potential name clash here?

Reply:
If you spin up multiple clusters using the tools, I believe these will clash. - i.e.: recall how you get "everything" back from the ec2.py tool. In your "hosts" file, you currently have entries like the following:
```
[masters:children]
tag_group_masters_aws
```
... this will then grab masters across all clusters that ec2.py returns unless we make these unique. 

Using the `env_id` aligns with how we do it for OpenStack and it ensures uniqueness.

Reply:
BTW: this can also be done as a follow-up PR, but we need it ASAP as we'll want to manage multiple clusters pretty much on day 1.

Reply:
@oybed you are right. What about adding the following to ec2.ini?

instance_filters = tag:env_id=

There is already an entry called 'regions = ' the user needs to modify, so we could add that as well. I think this could be easier than modify all the entries in the 'hosts' file, so we can maintain that file as it is for every single inventory